pos_slate/model/pos_session.py

```diff
@@ -0,0 +1,67 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    POS Keep Draft Orders module for Odoo
+#    Copyright (C) 2013-2014 GRAP (http://www.grap.coop)
+#    @author Julien WESTE
+#    @author Sylvain LE GAL (https://twitter.com/legalsylvain)
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp.exceptions import Warning
+from openerp.osv.orm import Model
+from openerp.tools.translate import _
+
+
+class pos_session(Model):
+    _inherit = 'pos.session'
+
+    def _remove_draft_orders(self, cr, uid, ids, context=None):
+        po_obj = self.pool['pos.order']
+        for ps in self.browse(cr, uid, ids, context=context):
+            for po in ps.order_ids:
+                # Check if there is a partial payment
+                if po.is_partial_paid:
+                    raise Warning(
+                        _("You cannot confirm this session, because '%s'"
+                            " is in a 'draft' state with payments.\n\n"
```

Comment:
is in a 'draft' state with payments
=> is still in 'draft' state with associated payments